pkg/apis/autoscaling/v1alpha1/pa_types.go

```diff
@@ -80,6 +80,9 @@ type PodAutoscalerSpec struct {
 	// is responsible for quickly right-sizing.
 	ScaleTargetRef corev1.ObjectReference `json:"scaleTargetRef"`
 
+	// Reachable specifies whether or not the `ScaleTargetRef` can be reached.
+	Reachable bool `json:"reachable"`
```

Comment:
I like the additional abstraction here a lot! Doesn't this belong in status though kinda? Or are we putting it in the spec because we'd have another controller mess with the PAs status?

Reply:
It's established by the Revision and passed down to the PodAutoscaler, which is why it landed in Spec.  🤷‍♂

Reply:
It just felt weird to me, I‘m not an API expert though. If you don‘t mind it I won‘t either

Reply:
👍 to me it feels strange also :)

Reply:
I think the confusion is coming from the name. 

When we see `pa.Reachable`, we're instinctually thinking "is the PA reachable?" which is something that should be in the Status. But that's not what this field is, it's actually "is the target resource reachable?", which suggests maybe we should change the name.

I'll propose `TargetReachable` in the separate PR (coming soon).

Reply:
Moved this change to a separate PR:  #5147